src/overlays/actors/ovl_En_Takara_Man/z_en_takara_man.c

```diff
@@ -27,25 +34,201 @@ const ActorInit En_Takara_Man_InitVars = {
     (ActorFunc)EnTakaraMan_Update,
     (ActorFunc)EnTakaraMan_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Takara_Man/EnTakaraMan_Destroy.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Takara_Man/EnTakaraMan_Init.s")
+static u8 sTakaraIsInitialized = false;
+
+extern SkeletonHeader D_06004FE0;
+extern AnimationHeader D_06000498;
+
+void EnTakaraMan_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void EnTakaraMan_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnTakaraMan* this = THIS;
+
+    if (sTakaraIsInitialized) {
+        Actor_Kill(&this->actor);
+        osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ もういてる原 ☆☆☆☆☆ \n" VT_RST); // "Already initialized"
+        return;
+    }
+
+    sTakaraIsInitialized = true;
+    osSyncPrintf("\n\n");
+    osSyncPrintf(VT_FGCOL(PURPLE) "☆☆☆☆☆ ばぅん！ ☆☆☆☆☆ %x\n" VT_RST,
+                 globalCtx->actorCtx.flags.chest); // "Bun! %x" (needs a better translation)
+    globalCtx->actorCtx.flags.chest = 0;
+    gSaveContext.dungeonKeys[gSaveContext.mapIndex] = -1;
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06004FE0, &D_06000498, this->limbDrawTbl, this->transitionDrawTbl,
+                     10);
+    thisx->posRot2.pos = thisx->posRot.pos;
+    this->pos = thisx->posRot.pos;
+    thisx->posRot.pos.x = 133.0f;
+    thisx->posRot.pos.y = -12.0f;
+    thisx->posRot.pos.z = 102.0f;
+    Actor_SetScale(&this->actor, 0.013f);
+    this->height = 90.0f;
+    this->originalRoomNum = thisx->room;
+    thisx->room = -1;
+    thisx->posRot.rot.y = thisx->shape.rot.y = -0x4E20;
+    thisx->unk_1F = 1;
+    this->actionFunc = func_80B176E0;
+}
+
+void func_80B176E0(EnTakaraMan* this, GlobalContext* globalCtx) {
+    f32 frameCount = SkelAnime_GetFrameCount(&D_06000498.genericHeader);
+
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06000498, 1.0f, 0.0f, (s16)frameCount, 0, -10.0f);
+    if (!this->unk_214) {
+        this->actor.textId = 0x6D; // "Open the chest and..Surprise! ... 10 Rupees to play .. Yes/No"
```

Comment:
i dont think we should be putting text comments in the code anymore. its something we used to do, but its not maintainable long term and clutters stuff up

Reply:
ez lookup here, idk where else the data also is available
https://github.com/NEstelami/ZAP2/blob/master/ZAP2/TextTable.h